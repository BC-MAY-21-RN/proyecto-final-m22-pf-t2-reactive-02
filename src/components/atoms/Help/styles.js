import {StyleSheet} from 'react-native';
import colors from '../../../const/colors';

const styles = StyleSheet.create({
  container: {
    width: 260,
    height: 400,
  },
  title: {
    color: colors.pink,
    fontSize: 16,
    fontWeight: 'bold',
    marginLeft: 10,
    marginTop: 10,
  },
  text: {
    fontSize: 16,
    textAlign: 'justify',
    maxWidth: 240,
    marginLeft: 10,
  },
  text2: {
    alignSelf: 'center',
    marginVertical: 5,
  },
});

export default styles;
