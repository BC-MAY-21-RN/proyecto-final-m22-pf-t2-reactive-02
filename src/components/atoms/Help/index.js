import React from 'react';
import {View, Text} from 'react-native';
import {Overlay} from 'react-native-elements/dist/overlay/Overlay';
import styles from './styles';
import texts from '../../../const/texts';

export default function Help({overlay}) {
  return (
    <Overlay
      isVisible={overlay.state}
      onBackdropPress={() => overlay.changeState(false)}>
      <View style={styles.container}>
        {texts.helpTexts.map((element, i) => (
          <View key={i}>
            <Text style={styles.title}>{element.title}</Text>
            <Text style={styles.text}>{element.text}</Text>
            {element.example !== undefined ? (
              <Text style={styles.text2}>{element.example}</Text>
            ) : null}
          </View>
        ))}
      </View>
    </Overlay>
  );
}
