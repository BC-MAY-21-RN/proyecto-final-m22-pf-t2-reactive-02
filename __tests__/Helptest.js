import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Help from '../src/components/atoms/Help';

describe('Test from Help component', () => {
  it('Help snapshot test', () => {
    const renderHelp = renderer.create(<Help overlay={false} />).toJSON();
    expect(renderHelp).toMatchSnapshot();
  });
});
