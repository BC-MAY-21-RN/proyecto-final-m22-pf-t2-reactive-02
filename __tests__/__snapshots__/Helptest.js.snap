// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test from Help component Help snapshot test 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, .4)",
        "height": "100%",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    testID="RNE__Overlay__backdrop"
  />
  <View
    pointerEvents="box-none"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 3,
          "padding": 10,
          "shadowColor": "rgba(0, 0, 0, .3)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowRadius": 4,
        }
      }
    >
      <View
        style={
          Object {
            "height": 400,
            "width": 260,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#DB6C9E",
                "fontSize": 16,
                "fontWeight": "bold",
                "marginLeft": 10,
                "marginTop": 10,
              }
            }
          >
            Nombre:
          </Text>
          <Text
            style={
              Object {
                "fontSize": 16,
                "marginLeft": 10,
                "maxWidth": 240,
                "textAlign": "justify",
              }
            }
          >
            El nombre debe de ser al menos un carácter
          </Text>
        </View>
        <View>
          <Text
            style={
              Object {
                "color": "#DB6C9E",
                "fontSize": 16,
                "fontWeight": "bold",
                "marginLeft": 10,
                "marginTop": 10,
              }
            }
          >
            Correo electrónico:
          </Text>
          <Text
            style={
              Object {
                "fontSize": 16,
                "marginLeft": 10,
                "maxWidth": 240,
                "textAlign": "justify",
              }
            }
          >
            El correo electrónico tiene que ser correcto ejemplo:
          </Text>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "marginVertical": 5,
              }
            }
          >
            petmyheart@gmail.com
          </Text>
        </View>
        <View>
          <Text
            style={
              Object {
                "color": "#DB6C9E",
                "fontSize": 16,
                "fontWeight": "bold",
                "marginLeft": 10,
                "marginTop": 10,
              }
            }
          >
            Contraseña:
          </Text>
          <Text
            style={
              Object {
                "fontSize": 16,
                "marginLeft": 10,
                "maxWidth": 240,
                "textAlign": "justify",
              }
            }
          >
            La contraseña tiene que tener al menos un carácter numérico, una letra mayúscula y una letra minúscula.
          </Text>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "marginVertical": 5,
              }
            }
          >
            Contraseña123
          </Text>
        </View>
        <View>
          <Text
            style={
              Object {
                "color": "#DB6C9E",
                "fontSize": 16,
                "fontWeight": "bold",
                "marginLeft": 10,
                "marginTop": 10,
              }
            }
          >
            Términos y condiciones:
          </Text>
          <Text
            style={
              Object {
                "fontSize": 16,
                "marginLeft": 10,
                "maxWidth": 240,
                "textAlign": "justify",
              }
            }
          >
            Se tienen que aceptar los términos y condiciones.
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
